hpx/util/register_locks.hpp

```diff
@@ -78,6 +78,23 @@ namespace hpx { namespace util
         void const* mtx_;
     };
 
+    template <typename Mutex>
+    struct ignore_mutex_while_checking
```

Comment:
Is this still needed? After all, a mutex is something lockable, and this isn't used now anymore.